File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    result=0
    for i in range (1,number+1):
        cnt=0
        for j in range (1,int(i**(1/2)+1)):
            if i%j==0:
                cnt+=1
                if j**2!=i:
                    cnt+=1
        if cnt>limit :
            result+=power
        else:
            result+=cnt
    return result

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    var result = 0;
    
    // 약수 개수를 저장하는 배열
    var divisors = new Array(number + 1).fill(0);

    for (let i = 1; i <= number; i++) {
        for (let j = i; j <= number; j += i) {
            // i의 배수에 대해 약수 개수를 증가
            divisors[j]++;
        }

        if (divisors[i] > limit) {
            result += power;
        } else {
            result += divisors[i];
        }
    }

    return result;
}
